src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -396,10 +398,37 @@ internal override async ValueTask ShutdownWriteCompleted(CancellationToken cance
             await _state.ShutdownWriteCompletionSource.Task.ConfigureAwait(false);
         }
 
-        internal override void Shutdown()
+        public override void CompleteWrites()
         {
             ThrowIfDisposed();
-            StartShutdown(QUIC_STREAM_SHUTDOWN_FLAGS.GRACEFUL, errorCode: 0);
+            try
+            {
+                StartShutdown(QUIC_STREAM_SHUTDOWN_FLAGS.GRACEFUL, errorCode: 0);
+            }
+            catch (Exception ex) when (ex is not OutOfMemoryException)
+            {
+                throw new IOException("Unable to complete writes: " + ex.Message, ex);
```

Comment:
We'll do a pass to move all resources into a resx subsequently?

Reply:
Yes, there's a QUIC issue for it.